src/guard.rs

```diff
@@ -31,10 +31,37 @@ pub struct SemaphoreGuardArc {
     panicking: bool,
 }
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum SemaphoreGuardSplitErr {
+    Underflow
+}
+
 impl<'a> SemaphoreGuard<'a> {
     pub fn new(semaphore: &'a Semaphore, amount: usize) -> Self {
         SemaphoreGuard { semaphore, amount, panicking: panicking() }
     }
+
+    /// Combine two `SemaphoreGuard`s into one, with the sum of the originals' permits.
+    ///
+    /// # Examples
+    /// ```
+    /// # use async_weighted_semaphore::Semaphore;
+    /// # tokio_test::block_on(async {
+    /// let semaphore = Semaphore::new(15);
+    /// let mut g1 = semaphore.acquire(10).await.unwrap();
+    /// let g2 = semaphore.acquire(5).await.unwrap();
+    /// g1.extend(g2);
+    /// # })
+    /// ```
+    pub fn extend(&mut self, other: SemaphoreGuard<'a>) {
+        if std::ptr::eq(self.semaphore, other.semaphore) {
+            self.amount += other.forget();
```

Comment:
Need to handle overflow here.

Reply:
Actually, it's handled automatically.  I'm not sure where, but somehow the semaphore gets poisoned on overflow.  See `test_extend_overflow`.

Reply:
The addition operation itself could overflow if extend is called multiple times 8 times with MAX_AVAILABLE.